string node_name
string pub_name
string frame_name